content/docs/troubleshooting/webhook.md

```diff
@@ -1,56 +1,991 @@
+in the Google documentation.
+
+### Cause 2: EKS on a custom CNI
+
+If you are on EKS and you are using a custom CNI such as Weave or Calico,
+the Kubernetes API server (which is in its own node) might not be able to
+reach the webhook pod. This happens because the control plane cannot be
+configured to run on a custom CNI on EKS, meaning that the CNIs cannot
+enable connectivity between the API server and the pods running in the
+worker nodes.
+
+Supposing that you are using Helm, the workaround is to add the following
+value in your `values.yaml` file:
+
+```yaml
+webhook:
+  hostNetwork: true
+  securePort: 10260
+```
+
+Or if you are using Helm from the command-line, use the following flag:
+
+```sh
+--set webhook.hostNetwork=true --set webhook.securePort=10260
+```
+
+By setting `hostNetwork` to `true`, the webhook pod will be run in the
+host's network namespace. By running in the host's network namespace, the
+webhook pod becomes accessible over the node's IP, which means you will
+work around the fact that kube-apiserver can't reach any pod IPs nor
+cluster IPs.
+
+By setting `securePort` to 10260 instead of relying on the default value
+(which is 10250), you will prevent a conflict between the webhook and the
+kubelet. The kubelet, which is an agent that runs on every Kubernetes
+worker node and that runs directly on the host, uses the port 10250 to
+expose its internal API to kube-apiserver.
+
+To understand how `hostnetwork` and `securePort` interact, we have to look
+at how the TCP connection is established. When the kube-apiserver process
+tries to connect to the webhook pod, kube-proxy (which also runs on control
+plane nodes, even without a CNI) kicks in and translates the webhook's
+cluster IP to the webhook's host IP:
+
+```diagram
+   https://cert-manager-webhook.cert-manager.svc:443/validate
+            |
+            |Step 1: resolve to the cluster IP
+            v
+   https://10.43.103.211:443/validate
+            |
+            |Step 2: send TCP packet
+            v
+   src: 172.28.0.1:43021
+   dst: 10.43.103.211:443
+            |
+            |Step 3: kube-proxy rewrite  (cluster IP to host IP)
+            v
+   src: 172.28.0.1:43021
+   dst: 172.28.0.2:10260
+            |
+            |                              control-plane node
+            |                           (host IP: 172.28.0.1)
+------------|--------------------------------------------------
+            |                           (host IP: 172.28.0.2)
+            v                                     worker node
+   +-------------------+
+   | webhook pod       |
+   | listens on        |
+   | 172.28.0.2:10260  |
+   +-------------------+
+```
+
+The reason 10250 is used as the default `securePort` is because it works
+around another limitation with GKE Private Clusters, as detailed in the
+above section [GKE Private Cluster](#gke-private-cluster).
```

Comment:
@wallrj I finally understand how this works!! I detailed how I think it works, let me know if that helps.